syntax = "proto3";

message Schema {
    string __version__ = 1;
}

message LogSchema {
    optional string __version__ = 1;
    optional int32 logID = 2;
    optional string log = 3;
    optional string logSource = 4;
    optional string hostname = 5;
}

message ParserSchema {
    optional string __version__ = 1;
    optional string parserType = 2;
    optional string parserID = 3;
    optional int32 EventID = 4;
    optional string template = 5;
    repeated string variables = 6;
    optional int32 parsedLogID = 7;
    optional int32 logID = 8;
    optional string log = 9;
    map<string, string> logFormatVariables = 10;
    optional int32 receivedTimestamp = 11;
    optional int32 parsedTimestamp = 12;
}

message DetectorSchema {
    optional string __version__ = 1;
    optional string detectorID = 2;
    optional string detectorType = 3;
    optional int32 alertID = 4;
    optional int32 detectionTimestamp = 5;
    repeated int32 logIDs = 6;
    optional float score = 8;
    repeated int32 extractedTimestamps = 9;
    optional string description = 10;
    optional int32 receivedTimestamp = 11;
    map<string, string> alertsObtain = 12;
}
